### PR TITLE
AndroidKeyStorage crash

### PR DESCRIPTION
**Pull request by biafra23**
Handles #1042 

Changes proposed in this pull request:
- Remove RTL "support" from app
- Temporarily change locale to allow key creation on Android M and earlier

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt [hunk 1]
```diff
@@ -231,7 +235,24 @@ class ApplicationModule(private val application: Application) {
...
     @Provides
     @Singleton
-    fun providesKeyStorage(@ApplicationContext context: Context): KeyStorage = AndroidKeyStorage(context)
+    fun providesKeyStorage(@ApplicationContext context: Context): KeyStorage =
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.M) {
+            val localeCopy = Locale.getDefault()
+            setLocale(context, Locale.ENGLISH)
+            val androidKeyStorage = AndroidKeyStorage(context)
+            setLocale(context, localeCopy)
+            androidKeyStorage
```
Comment (jpalvarezl) [inline app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt:244]: Are we doing a quick locale swap here just for the key generation step?

File: app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt [hunk 2]
```diff
@@ -231,7 +235,24 @@ class ApplicationModule(private val application: Application) {
...
+        } else {
+            AndroidKeyStorage(context)
+        }
+
+    private fun setLocale(context: Context, locale: Locale) {
```
Comment (jpalvarezl) [inline app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt:249]: maybe we could have a `private fun Context.setLocale()` method instead? Although it is not important.

File: app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt [hunk 3]
```diff
@@ -231,7 +235,24 @@ class ApplicationModule(private val application: Application) {
 
     @Provides
     @Singleton
-    fun providesKeyStorage(@ApplicationContext context: Context): KeyStorage = AndroidKeyStorage(context)
+    fun providesKeyStorage(@ApplicationContext context: Context): KeyStorage =
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.M) {
```
Comment (jpalvarezl) [inline app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt:239]: Do we need to do the API level check? I suppose we could just always use `Locale.ENGLISH` specifically for `KeyStorage` as well, then there would be fewer surprises.

File: app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt (hunk 1 above)
Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt:244]: Yes. That is the solution I fond on the internet :-/

File: app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt (hunk 3 above)
Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt:239]: Yes. We could. My idea was to remove this check and the branch, when have minSdkVersion > M.

File: app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt (hunk 1 above)
Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt:244]: this should be marked with //FIXME followed by a description of the reason for this workaround

Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt:245]: Added in fd5fbe51fb743768962bdba6212d18fe08007c17

File: app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt (hunk 3 above)
Comment (jpalvarezl) [reply app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt:240]: Would it be better to always use `Locale.English` independently of the SDK version for `bouncy_castle`? Having this small fork in logic here could be problematic as we would have small changes in behaviour depending on the device.

**Review by elgatovital — APPROVED** [review]

File: app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt (hunk 3 above)
Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt:240]: My risk assessment is like this: 
- If the locale switching turns out to be a problem, then it is only a problem on a minority of devices )if we have this branch logic)
- Google considered this a bug and fixed it, so it should not occur in devices after M

Comment (jpalvarezl) [reply app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt:240]: That makes sense. I understand now how we are better covered with this approach. Thanks!

**Review by jpalvarezl — APPROVED** [review]

**Comment by DmitryBespalov** [comment]
sweet! 🍬

**Pull request merged**